Sources/FluentKit/Query/Builder/QueryBuilder.swift

```diff
@@ -238,8 +243,46 @@ public final class QueryBuilder<Model>
             }
         }
         
+        let model = Model.init()
+        switch query.action {
+        case .delete:
+            if !self.forceDelete {
+                model.touchTimestamps(.delete, .update)
+                query.action = .update
+                query.input = [.dictionary(model.input.values)]
+            }
+        case .create:
+            var data: [DatabaseQuery.Value] = []
+            
+            for case .dictionary(var nested) in query.input {
+                if let createTimestamp = Model().timestamps.filter({ $0.trigger == .create }).first {
+                    nested[createTimestamp.path.first!] = .bind(Date())
+                }
+                if let updateTimestamp = Model().timestamps.filter({ $0.trigger == .update }).first {
+                    nested[updateTimestamp.path.first!] = .bind(Date())
+                }
+                data.append(.dictionary(nested))
+            }
+            
+            query.input = data
+        case .update:
+            var data: [DatabaseQuery.Value] = []
+            
+            for case .dictionary(var nested) in query.input {
+                if let updateTimestamp = Model().timestamps.filter({ $0.trigger == .update }).first {
```

Comment:
We could probably factor this out into a method since it's identical between both the create/update actions and on create/update timestamps.